.ci/run_tests.sh

```diff
@@ -21,8 +21,22 @@ else
 fi
 
 if [ "$TEST_SUITE" == "integration" ]; then
-    eval "$TEST_CMD" spec/02-integration/
+    if [ "$TEST_SPLIT" == "first" ]; then
+        # GitHub Actions, run first half of integration tests
+        eval "$TEST_CMD" $(ls -d spec/02-integration/* | head -n4)
+
+    elif [ "$TEST_SPLIT" == "second" ]; then
+        # GitHub Actions, run second half of integration tests
+        # In case of odd number of directories, the second half will be
+        # running one more directory than the first
+        eval "$TEST_CMD" $(ls -d spec/02-integration/* | tail -n+5)
```

Comment:
Not sure the comments match the behavior here... it mentions "first half" and "second half" and then comments about the odd number of directories, but this is really running the first 4 and then all the others (which are currently 6 folders).

Reply:
Still unconvinced about the arbitrary split here, we'll have to keep rememebering on which log the tests are.

Reply:
That comment came from the first iteration where the test cases were split in half evenly, except that did not result in a even split in terms of run time and that's why it was changed to this split that you see today which results in an almost even split of run time for those. If you can provide a better suggestion on the naming I would be more than happy to change the title.

Reply:
The biggest drawback about this is that in in order to to know whether an individual test goes into "first" and or "second" is to look at this file. I think this could be improved by an explicit renaming: `integration first half` could be renamed to `02-integration/01-04` and `integration second half` to `02-integration/05-xx`.

I see some alternatives, but they also have problems.

We could make the split directly on the filesystem (splitting `spec/02-integration/*` into two folders).
* Pro: To know what half your file is in, just look at the path. It is very "obvious".
* Pro: Impossible to forget to "assign a half" to a test.
* Con: Need "manually move files and folders from one folder to the other" from time to time in order to keep both halves "balanced".
* Con: Ties the "logical layout" of the tests to a "physical need" of one particular CI server.

We could add tags to the top `describe` on each file and use that for filtering.
* Pro: To know what "half" your file is in, just look at the top `describe`.
* Pro: "moving files from one half to the other" is a small diff - just retag.
* Pro: Allows for more granular splitting (by file instead of by folder). We could potentially split further (by test) but I don't think that's needed.
* Con: First commit would require manually tagging all files.
* Con: Need to remember to add the tag on files to keep things balanced.
* Con: Potentially execute `before` blocks that we don't need.
* Con: The "names" of each group of test would have to remain "nebulous", like "first" and "second", or "red" and "blue". 

If I had done this PR, I would have probably gone with the tags, since they look like a more natural fit to me. However the current approach has its own merits (provided we rename to something more obvious).